### ISSUE TITLE
Expose ApiClient's InteceptRequest And InterceptResponse part 2

### ISSUE THREAD

**Issue by john-do3** [XeroAPI/Xero-NetStandard#563]
**SDK you're using (please complete the following information):**
 - 9.2.0

**Is your feature request related to a problem? Please describe.**
In issue #240, ApiClient's InteceptRequest And InterceptResponse methods were made protected virtual - and reverted back to partial in https://github.com/XeroAPI/Xero-NetStandard/commit/c6b2b863dd9ea852ef4a82d7ab2f5dabc02988cc commit.

**Describe the solution you'd like**
Please make it virtual again.

**Comment by bruce-palmer** [comment]
Second this. I am using these methods for additional logging, and this regression has broken my code.

**Comment by LucasRodriguez0302** [comment]
I agree, we have been using this for logging purposes. Can we have this reverted?

**Comment by JamesGrantTradify** [comment]
Hey @vigneshk1121,

Just checking - do you have any update on when this issue might be addressed?
This is currently blocking us from updating the package and is leading to additional issues due to being on an outdated version.

Thank you!
